src/Intervals.php

```diff
@@ -0,0 +1,197 @@
+                return array('intervals' => array(array('start' => self::zero(), 'end' => $constraint)), 'devConstraints' => array());
+            }
+            if ($op === '!=') {
+                // convert !=x to intervals of 0 - <x && >x - +inf
+                return array('intervals' => array(
+                    array('start' => self::zero(), 'end' => new Constraint('<', $constraint->getVersion())),
+                    array('start' => new Constraint('>', $constraint->getVersion()), 'end' => self::positiveInfinity()),
+                ), 'devConstraints' => array());
+            }
+
+            // convert ==x to an interval of >=x - <=x
+            return array('intervals' => array(
+                array('start' => new Constraint('>=', $constraint->getVersion()), 'end' => new Constraint('<=', $constraint->getVersion())),
+            ), 'devConstraints' => array());
+        }
+
+        $constraints = $constraint->getConstraints();
+
+        $intervalGroups = array();
+        $dev = array();
+        foreach ($constraints as $c) {
+            $res = self::get($c);
+            if ($res['intervals']) {
+                $intervalGroups[] = $res['intervals'];
+            }
+            if ($res['devConstraints']) {
+                $dev = array_merge($dev, $res['devConstraints']);
+            }
+        }
+
+        if (count($intervalGroups) === 1) {
+            return array('intervals' => $intervalGroups[0], 'devConstraints' => $dev);
+        }
+
+        $borders = array();
+        foreach ($intervalGroups as $group) {
+            foreach ($group as $interval) {
+                $borders[] = array('version' => $interval['start']->getVersion(), 'operator' => $interval['start']->getOperator(), 'side' =>'start');
+                $borders[] = array('version' => $interval['end']->getVersion(), 'operator' => $interval['end']->getOperator(), 'side' =>'end');
+            }
+        }
+
+        $opSortOrder = self::$opSortOrder;
+        usort($borders, function ($a, $b) use ($opSortOrder) {
+            if ($a['version'] === $b['version']) {
+                return $opSortOrder[$a['operator']] - $opSortOrder[$b['operator']];
+            }
+
+            return version_compare($a['version'], $b['version']);
+        });
+
+        $activeIntervals = 0;
+        $intervals = array();
+        $activationThreshold = $constraint->isConjunctive() ? \count($intervalGroups) : 1;
+        $active = 0;
+        foreach ($borders as $border) {
+            if ($border['side'] === 'start') {
+                $activeIntervals++;
+            } else {
+                $activeIntervals--;
+            }
+            if (!$active && $activeIntervals >= $activationThreshold) {
+                $intervals[] = array('start' => new Constraint($border['operator'], $border['version']));
+                $active = $activeIntervals;
+            }
+            if ($active && $activeIntervals < $active) {
```

Comment:
Shouldn't this be `<$activationThreshold`? No need to store the number in `$active`?

Reply:
Yes true, got sidetracked with another bug at some point, but this isn't needed :)